### ISSUE TITLE
Adding a new project in VS with EnvDTE.Solution2.AddFromFile(string) doesn't raise Workspace.WorkspaceChanged event

### ISSUE THREAD

**Issue by twsouthwick** [dotnet/roslyn#17464]
**Version Used**: 
15.0.26228.0

**Steps to Reproduce**:

1. Create a project
2. Remove it from the solution (but it's still on disk)
3. Run the following code in VS:

``` csharp
var tcs = new TaskCompletionSource<bool>();

void WorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
{
    if (e.Kind == WorkspaceChangeKind.ProjectAdded)
    {
        tcs.TrySetResult(true);
    }
}

_workspace.WorkspaceChanged += WorkspaceChanged;

_solution.AddFromFile(pathToProject);

await tcs.Task.ConfigureAwait(false);

_workspace.WorkspaceChanged -= WorkspaceChanged;
```

where `_workspace` is of type `VisualStudioWorkspace`.

**Expected Behavior**:

The `WorkspaceChanged` event gets called and the `TaskCompletionSource` gets set. This was the behavior until I upgraded to this build today.

**Actual Behavior**:

The project gets added (the return of AddFromFile is a valid `Project` instance), but the event is never fired.

**Comment by twsouthwick** [comment]
@jasonmalinowski Could your change in commit d2f3957f34760926e2b8e05b8a85d47347413d4f be related to this?

**Comment by jasonmalinowski** [comment]
@twsouthwick: that change would only control eventing for the MiscellaneousFilesWorkspace I'd believe.

**Comment by twsouthwick** [comment]
Ok - didn't realize that was a different workspace.

This issue seems to be more far reaching than I first thought. I created a simple VSIX that just imported VisualStudioWorkspace and I'm seeing that not all projects get loaded in my VisualStudioWorkspace. I have  a project with 3 projects (1 CS Winforms, 1 CS console, and 1 VB Winforms) and only one is showing up as available in the Workspace.CurrentSolution.ProjectId

**Comment by jasonmalinowski** [comment]
@twsouthwick I bet you're seeing the issue @Pilchie fixed in #17335. As a check, if you open one of the .cs files in one of the projects that isn't in the ProjectIds collection, do you then see the ProjectId added? The bug is we failed to completely populate it in all cases.

Email me privately if you want to confirm which builds have this bug.

**Comment by twsouthwick** [comment]
Yup - that is the behavior I see. I'll go ahead and close this and if it ends up still repro'ing after the fix is available, I'll reopen it.
